001-finding_resources.md

```diff
@@ -0,0 +1,120 @@
+# Design Document 001 - Finding Resources
+
+TODO: Overview
+
+## Finding SDF URI Resources
+
+The following sections discuss how `<uri>` elements in SDF files are
+resovled in order to find meshes and models. This section does not cover
+finding other resources, such as plugins.
+
+### Current Process
+
+This section describes how Ignition Gazebo currently finds Mesh URI
+resources. SDF contains multiple instances of URI elements, but the current
+implementation of Gazebo handles only mesh URIs. The information contained
+here can be extrapolated to handle other types of resources, such as
+textures and audio files.
+
+A user of Ignition may not need or use Gazebo. If this describes you, then
+skip directly to the last step which describes how
+`ignition::common::findFile` can be used to resolve a resource.
```

Comment:
Maybe we could explain better what kind of user would be using SDF URIs directly with `ign-common`? I'm not sure if that's common usage :drum:

Reply:
Added more details in a3696f8e717265fb69c906bc630f83723b50d8dd